bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1731,31 +1743,29 @@ bool BCStateTran::onMessage(const CheckpointSummaryMsg *m, uint32_t msgLen, uint
   // if we have too many messages from the same replica
   if (numOfMsgsFromSender >= (psd_->getMaxNumOfStoredCheckpoints() + 1)) {
     LOG_WARN(logger_, "Too many messages from replica: " << KVLOG(replicaId, numOfMsgsFromSender));
-    return true;
+    return;
   }
 
-  auto p = summariesCerts.find(m->checkpointNum);
-  CheckpointSummaryMsgCert *cert = nullptr;
-
-  if (p == summariesCerts.end()) {
-    cert = new CheckpointSummaryMsgCert(
-        replicaForStateTransfer_, replicas_.size(), config_.fVal, config_.fVal + 1, config_.myReplicaId);
-    summariesCerts[m->checkpointNum] = cert;
-  } else {
-    cert = p->second;
+  if (summariesCerts.find(m->checkpointNum) == summariesCerts.end()) {
+    summariesCerts.emplace(
+        m->checkpointNum,
+        new CheckpointSummaryMsgCert(
+            replicaForStateTransfer_, replicas_.size(), config_.fVal, config_.fVal + 1, config_.myReplicaId));
   }
+  auto &cert = *summariesCerts.find(m->checkpointNum)->second;
```

Comment:
This is dangerous.
If you use summariesCerts.find, and then immidiately use cert - assert immidately after.
Split this into 2, access second after assertion.
Continuing without, might be an undefined behavior in case of a bug.

Reply:
done